src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -3412,5 +3414,14 @@ internal class TestingStuff
             internal bool StopIndex;
             internal Action<CompareExchangeCommandBase> ModifyCompareExchangeTimeout;
         }
+
+        public void LowMemory(LowMemorySeverity lowMemorySeverity)
+        {
+            FieldCache_Fields.DEFAULT.PurgeAllCaches();
```

Comment:
1. I think it would be better to have a dedicated LuceneCleaner and move it there, I do not like the mix
2. Is this thread safe? PurgeAllCaches can be done concurrently? Can it be executed when values from there are used?

Reply:
1. OK
2. Yes. It's under lock. And is thread-safe.
https://github.com/ravendb/lucenenet/blob/1a404988db55357e3d07dcd4c7815ce2ac689405/src/Lucene.Net/Search/FieldCacheImpl.cs#L80
https://github.com/ravendb/lucenenet/blob/1a404988db55357e3d07dcd4c7815ce2ac689405/src/Lucene.Net/Search/FieldCacheImpl.cs#L66

Reply:
done